### ISSUE TITLE
WeDo extension calculates tilt angle incorrectly

### ISSUE THREAD

**Issue by ericrosenbaum** [LLK/scratch-vm#2444]
### Expected Behavior

The LEGO WeDo extension blocks that use values from the tilt sensor should measure tilt angles in degrees.

### Actual Behavior

It appears that the WeDo sends values to Scratch from its tilt sensor that range from 0 to 255. These values do not correspond to an angle in degrees. They're probably time-averaged accelerometer values. 

One way you can see that this is the case is by holding the sensor at an angle where the tilt angle block reports 45, and seeing that it makes an angle to the ground that is higher than that.

### Notes

To calculate an accurate angle from these accelerometer values will require a bit of trigonometry. (there are some notes e.g. [here](https://www.digikey.com/en/articles/using-an-accelerometer-for-inclination-sensing)).

Since the WeDo extension reports tilt values clamped between -45 and 45 degrees, we shouldn't have the trouble that arises near 90 degrees. We seem to done the math for all that (if I understand correctly) in the Vernier extension, [here](https://github.com/LLK/scratch-vm/blob/58c460b16d73af85a73d4f3bb4d81a663b7fa42c/src/extensions/scratch3_gdx_for/index.js#L338-L394), but maybe this one will be simpler.

Worth noting that:
- The micro:bit extension has a very similar [problem](https://github.com/LLK/scratch-vm/issues/1918); we had a proposed [fix](https://github.com/LLK/scratch-vm/pull/1913) on the firmware end but decided not to use it (I think due to version incompatibility problems, e.g. the fixed extension would report incorrect results with the old firmware, and vice versa). So we could still use a fix using trig on the extension side for that one.
- The LEGO BOOST firmware computes the angles, so we don't have this problem in that extension
- The LEGO EV3 extension does not contain blocks for measuring tilt angles, so no problem there

**Comment by daikifukumori** [comment]
I do not think that Scratch calculates tilt angle "incorrectly".
In my understanding, Scratch just [reads the value](https://github.com/LLK/scratch-vm/blob/58c460b16d73af85a73d4f3bb4d81a663b7fa42c/src/extensions/scratch3_wedo2/index.js#L780-L781) of BLE data. The unit of the data is already in degrees. For example, when the angle of the tilt sensor is 30 degrees, the BLE data sent out from the tilt sensor has 30 and the Scratch block "tilt angle" returns 30. (I measured it with a protractor!)(*)
However, the tilt sensor of WeDo 2.0 can measure the angle up to only 45 degrees (due to a physical structural limitation, probably). Thus, even when the angle is 60 degrees, the tilt sensor sends out the value 45. Likewise, when the angle is 90 degrees, the tilt sensor sends out the value 45 as well.
To sum up, this is an accuracy limitation of the tilt sensor.

*: The value of BLE data is a signed 8-bit integer. For example, 0xd3 is not 211(10), but -45(10).

**Comment by ericrosenbaum** [comment]
thanks for the explanation @daikifukumori, and sorry that we seem to have gotten confused about this!
